dirty_cat/_fuzzy_join.py

```diff
@@ -164,6 +171,13 @@ def fuzzy_join(
             f"analyzer should be either 'char', 'word' or 'char_wb', got {analyzer!r}",
         )
 
+    if encoder not in ["hashing", "count"]:
+        # TODO: implement check for generic vectorizer
+        raise ValueError(
+            "encoder should be either 'hashing', 'count', or a vectorizer object, got"
+            f" {encoder!r}"
+        )
```

Comment:
Should add an if statement with something like `if issubclass(encoder.__class__, _VectorizerMixin)`